main.cpp

```diff
@@ -234,15 +266,23 @@ int main(int argc, char *argv[]) {
 			}
 		}
 
-		// TODO: Remove these when connecting the pixel display to the emulator core.
-		pixels[pixel_index] ^= 0x00FFFFFF;
-		pixel_index = (pixel_index + 7) % (WIDTH * HEIGHT);
+		int cyclesThisFrame = 0;
+
+		while (cyclesThisFrame < MAX_CYCLES_PER_FRAME) {
+			int cycles = cpu.step();
+			cyclesThisFrame += cycles;
+			timer.step(cycles);
+			ppu.step(cycles);
+		}
+		
+		memcpy(pixels, ppu.getFrameBuffer(), NUM_PIXELS);
 
 		SDL_RenderClear(renderer);
-		SDL_UpdateTexture(texture, nullptr, pixels, WIDTH * sizeof(Uint32));
+		SDL_UpdateTexture(texture, nullptr, pixels, WIDTH * sizeof(u8) * 3);
 		SDL_RenderCopy(renderer, texture, nullptr, nullptr);
 		SDL_RenderPresent(renderer);
 
 		SDL_Delay(1000 / FPS);
 	}
+	return 0;
```

Comment:
Another complaint from Mingw was missing return, even though it's `main` and implicit. Oh well.